docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/update-folder-structure.md

```diff
@@ -0,0 +1,356 @@
+---
+title: v4 Plugin Migration - Update folder structure - Strapi Developer Docs
+description:
+canonicalUrl:
+sidebarDepth: 2
+---
+
+<!-- TODO: update SEO -->
+
+# v4 plugin migration: Update the folder structure
+
+!!!include(developer-docs/latest/update-migration-guides/migration-guides/v4/snippets/plugin-migration-intro.md)!!!
+
+::: callout
+Strapi v3 plugins required a specific folder structure.
+
+In Strapi v4, plugins are developed using a programmatic API, which gives flexibility in the folder structure.
+:::
+
+The folder structure of a Strapi v4 plugin should meet 2 requirements:
+
+- The root of the plugin folder should include a `strapi-server.js` and a `strapi-admin.js` entry files, for the [Server API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md) and [Admin Panel API](/developer-docs/latest/setup-deployment-guides/configurations/optional/environment.md), respectively.
```

Comment:
If there is no front-end then `strapi-admin.js` is not required, same for the backend. How can we say that there are optional?